docs/programming-guide.md

```diff
@@ -1005,7 +1068,8 @@ for details.
 </tr>
 <tr>
   <td> <b>foreach</b>(<i>func</i>) </td>
-  <td> Run a function <i>func</i> on each element of the dataset. This is usually done for side effects such as updating an accumulator variable (see below) or interacting with external storage systems. </td>
+  <td> Run a function <i>func</i> on each element of the dataset. This is usually done for side effects such as updating an accumulator variable (see below) or interacting with external storage systems. 
+  <br /><b>Note</b>: modifying variables outside of the <code>foreach()</code> will only have the desired effect when running locally (e.g. the spark shell) or when updating an accumulator. See <a href="#ClosuresLink">Understanding closures </a> for more details.</td>
```

Comment:
I think this should still be updated, and should probably make a few edits like spark -> Spark, or accumulator -> `Accumulator`